File: BOOM/Boom.Model/Bomb.cs
namespace Boom.Model
{
    //Fires nine shots covering the 3x3 grid of squares centred on the given row, column.
    public class Bomb : IWeapon
    {
        private int blastRadius = 1;

        public void Fire(int col, int row, GameBoard Board)
        {
            for (int startCol = col - blastRadius; startCol <= col + blastRadius; startCol++)
            {
                for (int startRow = row - blastRadius; startRow <= row + blastRadius; startRow++)
                {
                    if (startCol >= 0 && startCol < 10 && startRow >= 0 && startRow < 10)
                    {
                        Board.CheckSquareAndRecordOutcome(startCol, startRow);
                    }
                }
            }
        }
    }
}

File: BOOM.VB/Boom.Model/Bomb.vb

Namespace Boom.Model
	'Fires nine shots covering the 3x3 grid of squares centred on the given row, column.
	Public Class Bomb
		Implements IWeapon
		Private blastRadius As Integer = 1

        Public Sub Fire(col As Integer, row As Integer, Board As GameBoard) Implements IWeapon.Fire
            For startCol As Integer = col - blastRadius To col + blastRadius
                For startRow As Integer = row - blastRadius To row + blastRadius
                    If True Then
                        If startCol >= 0 AndAlso startCol < 10 AndAlso startRow >= 0 AndAlso startRow < 10 Then
                            Board.CheckSquareAndRecordOutcome(startCol, startRow)
                        End If
                    End If
                Next
            Next
        End Sub
    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================
